src/v/kafka/server/group_manager.cc

```diff
@@ -1570,14 +1571,22 @@ group_manager::list_groups() const {
     std::vector<listed_group> groups;
     for (const auto& it : _groups) {
         const auto& g = it.second;
+        if (
+          !filter_data.states_filter.empty()
+          && !filter_data.states_filter.contains(g->state())) {
```

Comment:
nit: i'd guess there probably isn't a real benefit to checking if it is empty so you could simplify the predicate

Reply:
If the states set is empty (which can happen for older version requests or newer version requests without a state filter specified) we want to include every group. So this is not just a performance shortcut. I should probably add a comment about this though to make it more apparent.

Reply:
>probably add a comment about this

good idea. i had to stare at it for a minute or two to get the point, wound up deleting a comment almost identical to Noah's.

Reply:
> If the states set is empty

ahh! of course, my mistake. thanks